pkg/queue/bounded_queue.go

```diff
@@ -105,12 +105,13 @@ func (q *BoundedQueue) Produce(item interface{}) bool {
 		return false
 	}
 
+	q.size.Add(1)
```

Comment:
This actually looks like a real bug, we are subtracting in both code branches, but adding in only one. Good catch!